include/swift/AST/Types.h

```diff
@@ -4065,6 +4083,10 @@ class SILFunctionType final : public TypeBase, public llvm::FoldingSetNode,
     return hasResultCache() ? 2 : 0;
   }
 
+  size_t numTrailingObjects(OverloadToken<SubstitutionMap>) const {
+    return hasPatternSubstitutions() + hasInvocationSubstitutions();
```

Comment:
It seems a bit puzzling to see two booleans being added. Could you add explicit casts here to make it clear that addition is being performed with both LHS and RHS being non-boolean integral type?